packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -243,6 +248,82 @@ export class CdkToolkit {
     }
   }
 
+  public async watch(options: WatchOptions) {
+    const rootDir = path.dirname(path.resolve(PROJECT_CONFIG));
+    debug("root directory used for 'watch' is: %s", rootDir);
+
+    const watchSettings: { include?: string | string[], exclude: string | string [] } | undefined =
+        this.props.configuration.settings.get(['watch']);
+    if (!watchSettings) {
+      throw new Error("Cannot use the 'watch' command without specifying at least one directory to monitor. " +
+        'Make sure to add a "watch" key to your cdk.json');
+    }
+
+    // For the "include" subkey under the "watch" key, the behavior is:
+    // 1. No "watch" setting? We error out.
+    // 2. "watch" setting without an "include" key? We default to observing "./**".
+    // 3. "watch" setting with an empty "include" key? We default to observing "./**".
+    // 4. Non-empty "include" key? Just use the "include" key.
+    const watchIncludes = this.patternsArrayForWatch(watchSettings.include, rootDir, /* returnRootDirIfEmpty */ true);
+    debug("'include' patterns for 'watch': %s", watchIncludes);
+
+    // For the "exclude" subkey under the "watch" key,
+    // the behavior is to add some default excludes in addition to the ones specified by the user:
+    // 1. The CDK output directory.
+    // 2. Any file whose name starts with a dot.
+    // 3. Any directory's content whose name starts with a dot.
+    // 4. Any node_modules and its content.
+    const outputDir = this.props.configuration.settings.get(['output']);
+    const watchExcludes = this.patternsArrayForWatch(watchSettings.exclude, rootDir, /* returnRootDirIfEmpty */ false).concat(
+      `${rootDir}/${outputDir}/**`,
+      `${rootDir}/**/.*`,
+      `${rootDir}/**/.*/**`,
+      `${rootDir}/**/node_modules/**`,
```

Comment:
If we have node_modules here, then we should likely have some defaults for other languages like `target/` for java and `.menv` for python...

Reply:
So, the reason for excluding `node_modules` specifically is that our `aws-cdk` CLI is a NodeJS package. Which means, you might have a `node_modules` directory even if you're using a non-JS/TS language (if you want to version the CLI package alongside your library packages).

I realized I should have put that in the comments. I've fixed it now.

Reply:
In non JavaScript scenarios users usually install the CLI globally since they don't really maintain a package.json file.

Reply:
Yes, **usually** being the crucial word here 😃.

Do you foresee any problems with excluding `node_modules` from our watching?

Reply:
Mmmm. I guess it's okay.